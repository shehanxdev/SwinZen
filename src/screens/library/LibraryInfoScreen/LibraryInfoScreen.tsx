import { useNavigation } from '@react-navigation/native';
import React, { useLayoutEffect } from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextAlignment, TextVariant } from '@sz/constants';

import { BaseScreen } from '../../components';

interface Points {
  title: string;
  body: string;
}

interface LibraryInfoProps {
  headerTitle: string;
  withIndex: boolean;
  content: {
    description: string;
    points: Array<Points>;
  };
}

export function LibraryInfoScreen({ route }) {
  const data = route.params.params as LibraryInfoProps;

  const navigation = useNavigation();

  // custom action for navigation header back button
  useLayoutEffect(() => {
    navigation.setOptions({
      title: data.headerTitle,
    });
  });

  return (
    <BaseScreen testID="LibraryInfoScreenTestID">
      <View style={tw`mx-4 mt-8`}>
        {data.content.description.length > 0 && (
          <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
            {data.content.description}
          </Text>
        )}
        {data.withIndex &&
          data.content.points?.map((point, index) => (
            <View key={point.title} style={tw`flex-1 ml-2`}>
              <View style={tw`flex-row gap-3`}>
                <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
                  {index + 1}.
                </Text>
                <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
                  {point.title}
                </Text>
              </View>
              <View style={tw`ml-6`}>
                <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
                  {point.body}
                  {`\n`}
                </Text>
              </View>
            </View>
          ))}
        {!data.withIndex &&
          data.content.points?.map(point => (
            <View key={point.title}>
              <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
                {point.title}
              </Text>
              <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
                {point.body}
              </Text>
            </View>
          ))}
      </View>
    </BaseScreen>
  );
}
