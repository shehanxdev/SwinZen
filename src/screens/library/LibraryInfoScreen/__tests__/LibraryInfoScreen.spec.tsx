import { render } from '@testing-library/react-native';
import * as React from 'react';

import { LibraryInfoScreen } from '../LibraryInfoScreen';

describe('Library Info Screen', () => {
  const testID = 'LibraryInfoScreenTestID';

  const mockRoute = {
    params: {
      params: {
        withIndex: true,
        content: {
          description: 'dummy description',
          points: [{ title: 'dummy title', body: 'dummy body' }],
        },
      },
    },
  };

  const getRenderedScreen = () => render(<LibraryInfoScreen route={mockRoute} />);

  it(`should render Library Info Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the Library Info Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
