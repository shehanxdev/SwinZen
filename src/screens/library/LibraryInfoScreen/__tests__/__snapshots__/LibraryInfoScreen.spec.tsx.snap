// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library Info Screen should render Library Info Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
  testID="LibraryInfoScreenTestID"
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "opacity": 0.1,
        "position": "absolute",
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              {
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 32,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              dummy description
            </Text>
            <View
              style={
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 8,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "gap": 12,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "lineHeight": 28,
                      },
                      {
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 18,
                        "fontWeight": "normal",
                        "textAlign": "auto",
                      },
                    ]
                  }
                >
                  1
                  .
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "lineHeight": 28,
                      },
                      {
                        "fontFamily": "SourceSansPro-SemiBold",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "auto",
                      },
                    ]
                  }
                >
                  dummy title
                </Text>
              </View>
              <View
                style={
                  {
                    "marginLeft": 24,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "lineHeight": 28,
                      },
                      {
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 18,
                        "fontWeight": "normal",
                        "textAlign": "auto",
                      },
                    ]
                  }
                >
                  dummy body
                  

                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
